src/main/scala/com/fulcrumgenomics/vcf/MakeTwoSampleMixtureVcf.scala

```diff
@@ -0,0 +1,200 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.vcf
+
+import java.util
+import java.util.Collections
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.Io
+import com.fulcrumgenomics.vcf.MakeTwoSampleMixtureVcf._
+import dagr.sopt
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.util.IntervalList
+import htsjdk.variant.variantcontext._
+import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriter, VariantContextWriterBuilder}
+import htsjdk.variant.vcf._
+
+import scala.collection.JavaConversions.asJavaCollection
+import scala.collection.JavaConversions.iterableAsScalaIterable
+import scala.collection.mutable.ListBuffer
+
+object MakeTwoSampleMixtureVcf {
+  val AlleleFractionField = "AF"
+  val GermlineFilter      = "alt_allele_in_normal"
+  val UnknownGtFilter     = "unknown_gt"
+  val MultiAllelicFilter  = "multi_allelic"
+
+  val TumorSampleName  = "tumor"
+  val NormalSampleName = "normal"
+}
+
+/**
+  * Creates a VCF by mixing two germline samples at a given proportion.
+  *
+  * @author Tim Fennell
+  */
+@sopt.clp(group=ClpGroups.VcfOrBcf, description=
+  """
+    |Creates a VCF by in-silico mixing genotypes from two samples. Writes a new VCF for a 'tumor' and
+    |optionally a 'normal' where the tumor is a mixture of two input samples, and the normal
+    |is one of the same input samples. Only outputs loci that are variant in the 'tumor'.
+    |Writes the 'AF' attribute in the format field with the expected allele fraction in the tumor.
+    |
+    |All loci (potentially restricted by intervals) that are variant in one or both samples are written
+    |to the output VCF, though in several cases the variants will be filtered:
+    |  - If either of the tumor or normal sample is no-called the resulting locus will have the
+    |    'unknown_gt' filter applied
+    |  - If the tumor and the normal have more than one alternative allele between them the
+    |    'multi_allelic' filter will be applied
+    |  - In tumor/normal mode (as opposed to tumor-only) loci that have an alt allele in the normal
+    |    sample will have the 'alt_allele_in_normal' filter applied
+  """
+)
+class MakeTwoSampleMixtureVcf
+( @arg(flag="i", doc="Input VCF file.")  val input: PathToVcf,
+  @arg(flag="o", doc="Output VCF file.") val output: PathToVcf,
+  @arg(flag="t", doc="Name of the 'tumor' sample in the input VCF.")  val tumor: String,
+  @arg(flag="n", doc="Name of the 'normal' sample in the input VCF.") val normal: String,
```

Comment:
Oh, I see, the input VCF is multi-sample.  This was not clear that both input samples should be in the single input VCF, rather than providing two input single-sample VCFs.